src/sdltiles.cpp

```diff
@@ -255,6 +259,45 @@ void init_interface()
 //***********************************
 //Non-curses, Window functions      *
 //***********************************
+void generate_alt_rect_texture()
+{
+#if SDL_BYTEORDER == SDL_BIG_ENDIAN
+    static const Uint32 rmask = 0xff000000;
+    static const Uint32 gmask = 0x00ff0000;
+    static const Uint32 bmask = 0x0000ff00;
+    static const Uint32 amask = 0x000000ff;
+#else
+    static const Uint32 rmask = 0x000000ff;
+    static const Uint32 gmask = 0x0000ff00;
+    static const Uint32 bmask = 0x00ff0000;
+    static const Uint32 amask = 0xff000000;
+#endif
+    SDL_Surface_Ptr altsurf( SDL_CreateRGBSurface( 0, 1, 1, 32, rmask, gmask, bmask, amask ) );
+    SDL_FillRect( altsurf.get(), NULL, SDL_MapRGB( altsurf->format, 255, 255, 255 ) );
+    SDL_Texture_Ptr alternate_rect_tex( SDL_CreateTextureFromSurface( renderer.get(), altsurf.get() ) );
```

Comment:
You need to remove SDL_Texture_Ptr here and replace with `alternate_rect_tex.reset( SDL_CreateTextureFromSurface( renderer.get(), altsurf.get() ) )`. Otherwise, it is creating a new pointer for the duration of `generate_alt_rect_texture()`, and the texture is lost after leaving this function. The overwriting effect you see is the null texture skipping the SDL draw calls.

Reply:
Thanks. I've fixed that.